src/Qir/Runtime/lib/Simulators/FullstateSimulator.cpp

```diff
@@ -408,7 +419,109 @@ namespace Quantum
 
             return (std::abs(actualProbability - probabilityOfZero) < precision);
         }
-    };
+
+      private:
+        std::ostream& GetOutStream(const void* location, std::ofstream& outFileStream);
+        void DumpMachineImpl(std::ostream& outStream);
+        void DumpRegisterImpl(std::ostream& outStream, const QirArray* qubits);
+        void GetStateTo(TDumpLocation location, TDumpToLocationCallback callback);
+        bool GetRegisterTo(TDumpLocation location, TDumpToLocationCallback callback, const QirArray* qubits);
+
+      private:
+        TDumpToLocationCallback const dumpToLocationCallback = [](size_t idx, double re, double im, TDumpLocation location) -> bool
+            {
+                std::ostream& outStream = *reinterpret_cast<std::ostream*>(location);
+
+                if (re != 0 || im != 0)
+                {
+                    outStream << "|" << std::bitset<8>(idx) << ">: " << re << "+" << im << "i" << std::endl;
+                }
+                return true;
+            };
+    }; // class CFullstateSimulator
+
+    void CFullstateSimulator::DumpMachineImpl(std::ostream& outStream)
+    {
+        outStream << "*********************" << std::endl;

```

Comment:
Are these rows of asterisks needed? I don't believe we have similar in any of our other diagnostic output functionality, but would need to double-check.

Reply:
(This has been copied from `DumpState()`)

Changed to be similar to what C# outputs.
C#: `channel($"# wave function for qubits with ids (least to most significant): {string.Join(";", ids)}");`